compiler/luci/pass/src/FuseTransposeWithMeanPass.cpp

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/FuseTransposeWithMeanPass.h"
+
+#include <luci/IR/CircleNode.h>
+#include <luci/Profile/CircleNodeOrigin.h>
+#include <luci/Service/Nodes/CircleConst.h>
+
+namespace
+{
+
+/**
+ *  Create a const for fused reduction indices
+ *
+ *  BEFORE
+ *                  |
+ *          [CircleTranspose, perm<0, 2, 3, 1>]
+ *                  |
+ *          [CircleMean, axis<3>]
+ *                  |
+ *
+ *  AFTER
+ *                  |
+ *          [CircleMean, axis<1>]       [CircleTranspose, perm<0, 2, 3, 1>]
+ *                  |                            |
+ *                                      [CircleMean, axis<3>]
+ *
+ */
+luci::CircleConst *create_fused_indices(luci::CircleConst *rindices,
+                                        const std::vector<uint32_t> &fused_rindices)
+{
+  assert(rindices != nullptr); // FIX_CALLER_UNLESS
+
+  if (rindices->dtype() != loco::DataType::S32)
+    return nullptr;
+
+  assert(fused_rindices.size() == rindices->size<loco::DataType::S32>());
+
+  auto fused_rindices_const = luci::clone(rindices);
+  auto name = rindices->name();
+  assert(name.length() > 0); // FIX_CALLER_UNLESS
+  fused_rindices_const->name(name + "_orig");
```

Comment:
Maybe you mean 
```suggestion
  fused_rindices_const->name(name + "_fused");
```
or something like this